src/pages/events/index.svelte

```diff
@@ -1,90 +1,164 @@
 <script>
-import EventFilters from 'components/EventFilters.svelte'
-import FilteredDisplay from 'components/FilteredDisplay.svelte'
-import { isParticipant, populateEventFilterFrom } from 'data/eventFiltering.js'
-import { join, events, loading } from 'data/events.js'
-import { me } from 'data/user.js'
-
-import Icon from 'fa-svelte'
-import { faExternalLinkAlt, faEdit } from '@fortawesome/free-solid-svg-icons'
-import { goto, params } from '@roxi/routify'
-import { flip } from 'svelte/animate';
-import { fade } from 'svelte/transition';
-
-const format = date => new Date(date).toLocaleDateString([], {
-  month: 'short',
-  day: 'numeric',
-  year: 'numeric',
-  timeZone: 'UTC',
-})
-const logoUrl = event => event.image_file?.url || ''
-
-let eventFilter = {}
-
-$: eventFilter = populateEventFilterFrom($params, $me)
+  import EventFilters from "components/EventFilters.svelte";
+  import FilteredDisplay from "components/FilteredDisplay.svelte";
+  import {
+    isParticipant,
+    populateEventFilterFrom,
+  } from "data/eventFiltering.js";
+  import { join, events, loading } from "data/events.js";
+  import { me } from "data/user.js";
+
+  import Icon from "fa-svelte";
+  import { faExternalLinkAlt, faEdit } from "@fortawesome/free-solid-svg-icons";
+  import { goto, params } from "@roxi/routify";
+  import * as animateScroll from "svelte-scrollto";
+  import { flip } from "svelte/animate";
+  import { fade } from "svelte/transition";
+
+  const format = (date) =>
+    new Date(date).toLocaleDateString([], {
+      month: "short",
+      day: "numeric",
+      year: "numeric",
+      timeZone: "UTC",
+    });
+  const logoUrl = (event) => event.image_file?.url || "";
+
+  let eventFilter = {};
+  // let scrollTo = $params.scrollTo
+
+  $: eventFilter = populateEventFilterFrom($params, $me);
+
+  function scrollToEvent(filteredEvents) {
+    if ($params.scrollTo) {
+      let eventNum;
+
+      filteredEvents.forEach((val, i) => {
+        if (val.id == $params.scrollTo) eventNum = i + 1;
+      });
+
+      if (!eventNum) return;
+
+      let eventId = `#event-${eventNum}`;
+
+      setTimeout(() => {
+        animateScroll.scrollTo({
+          element: eventId,
+          duration: 1000,
+        });
+      }, 1000);
+
+      // if the 'scrollTo' url parameter should be removed from the url, uncomment this and the 'scrollTo'
```

Comment:
I think the url parameter is fine for now. I appreciate the option to do it using `replaceState` though. So this commented code can be removed.